crates/outline_panel/src/outline_panel.rs

```diff
@@ -4112,18 +4269,62 @@ impl OutlinePanel {
                     continue;
                 }
 
+                let mut last_depth_at_level: Vec<Option<Range<Anchor>>> = vec![None; 10]; // Track last outline at each depth level
+
                 for outline in excerpt.iter_outlines() {
-                    self.push_entry(
-                        state,
-                        track_matches,
-                        PanelEntry::Outline(OutlineEntry::Outline(OutlineEntryOutline {
-                            buffer_id,
-                            excerpt_id,
-                            outline: outline.clone(),
-                        })),
-                        outline_base_depth + outline.depth,
-                        cx,
-                    );
+                    let outline_entry = OutlineEntryOutline {
+                        buffer_id,
+                        excerpt_id,
+                        outline: outline.clone(),
+                    };
+
+                    // Check if this outline should be skipped due to a collapsed parent
+                    let mut should_skip = false;
+                    for depth in 0..outline.depth {
```

Comment:
I'm not sure what's up with this extra sub-loop?
We seem to need to go over each outline, track which is folded and, while folded and has a larger depth than the folded one, we skip such outlines; stop skipping when the depth becomes not larger.
Additionally, check the ranges before folding and stop skipping if they do not overlap with the parent.

Did I miss something?